apps/ledger-live-desktop/src/renderer/modals/Receive/index.tsx

```diff
@@ -1,61 +1,99 @@
-import React, { PureComponent } from "react";
+import React, { useState, useCallback, useEffect } from "react";
 import logger from "~/renderer/logger";
 import Modal from "~/renderer/components/Modal";
 import Body, { StepId } from "./Body";
+import { useDispatch, useSelector } from "react-redux";
+import { accountsSelector } from "~/renderer/reducers/accounts";
+import { openModal, closeModal } from "~/renderer/actions/modals";
+
 type State = {
   stepId: StepId;
   isAddressVerified: boolean | undefined | null;
   verifyAddressError: Error | undefined | null;
 };
+
 const INITIAL_STATE = {
   stepId: "account" as StepId,
   isAddressVerified: null,
   verifyAddressError: null,
 };
-class ReceiveModal extends PureComponent<{}, State> {
-  state = INITIAL_STATE;
-  handleReset = () =>
-    this.setState({
-      ...INITIAL_STATE,
-    });
-
-  handleStepChange = (stepId: StepId) =>
-    this.setState({
-      stepId,
-    });
-
-  handleChangeAddressVerified = (isAddressVerified?: boolean | null, err?: Error | null) => {
+const ReceiveModal = () => {
+  const [state, setState] = useState<State>(INITIAL_STATE);
+
+  // You can access state properties like this:
```

Comment:
ChatGPT prompt? :p

Reply:
To convert from Class comp to functionnal comp yes ! Cleaning up the comments tho